RecoLocalTracker/SiStripClusterizer/plugins/ClustersFromRawProducer.cc

```diff
@@ -308,10 +317,16 @@ try {
     
 
     // If Fed hasnt already been initialised, extract data and initialise
-    if (!done[fedId]) { buffers[fedId].reset(fillBuffer(fedId, rawColl)); done[fedId]=true;}
-    auto buffer = buffers[fedId].get();
-    if unlikely(!buffer) continue;
-    
+    sistrip::FEDBuffer * buffer = done[fedId];
+    if (!buffer) { 
+      buffer = fillBuffer(fedId, rawColl).release();
+      if (!buffer) { continue;}
+      sistrip::FEDBuffer * exp = nullptr;
+      if (done[fedId].compare_exchange_strong(exp, buffer)) buffers[fedId].reset(buffer);
```

Comment:
Technically, this is still a race condition since there is no memory barrier applied after the call to ```buffers[fedId].reset(buffer)``` and therefore that change is not guaranteed to by synchronized across cores before a potential call to the destructor (where the value would be read). From a practical stand point given the TBB task will finish and flush the memory before the destructor would be called I think this work in this case.
I know it is less elegant, but you could just drop ```buffers``` and use ```done``` to manage the memory.